### ISSUE TITLE
Basic auth does not log username on failure

### ISSUE THREAD

**Issue by youtous** [containous/traefik#6722]
Hi, 
Is this behavior intended?

### Bug report

Basic auth middleware does not log provided username when authentication failed.

### What did you do?

1. Existing user, bad password: username IS NOT present in the log
```
$ curl http://test:testdsds@traefik.localhost.dv/user-exists-badpass
> 172.26.0.1 - - [20/Apr/2020:14:09:03 +0000] "GET /user-exists-badpass HTTP/1.1" 401 17 "-" "-" 5 "whoami@docker" "-" 1ms
```
`172.26.0.1 - -` **please note no username is logged**

2. Existing user, good password: username IS present in the log
```
$ curl http://test:test@traefik.localhost.dv/user-exists-goodpass
> 172.26.0.1 - test [20/Apr/2020:14:07:08 +0000] "GET /user-exists-goodpass HTTP/1.1" 200 396 "-" "-" 3 "whoami@docker" "http://172.26.0.2:80" 9ms
```
3. Unknown user: username IS NOT present in the log
```
$ curl http://noexist:badpass@traefik.localhost.dv/no-exists
> 172.26.0.1 - - [20/Apr/2020:14:14:49 +0000] "GET /no-exists HTTP/1.1" 401 17 "-" "-" 1 "whoami@docker" "-" 0ms
```

### What did you expect to see?

According https://github.com/fail2ban/fail2ban/issues/2693#issuecomment-618257204, the username should be logged when the authentication failed.

Excepted log:
```
$ curl http://test:test@traefik.localhost.dv/user-exists-goodpass
> 172.26.0.1 - test [20/Apr/2020:14:14:49 +0000] "GET /no-exists HTTP/1.1" 401 17 "-" "-" 1 "whoami@docker" "-" 0ms
```

### What did you see instead?

```
$ curl http://test:test@traefik.localhost.dv/user-exists-goodpass
> 172.26.0.1 - - [20/Apr/2020:14:14:49 +0000] "GET /no-exists HTTP/1.1" 401 17 "-" "-" 1 "whoami@docker" "-" 0ms
```

### Output of `traefik version`: 

Tested with traefik 1.7 and traefik 2.2: same behavior observed.

Configuration used for tests is available at https://github.com/fail2ban/fail2ban/issues/2693#issuecomment-616583684

**Comment by tompson** [comment]
This was already an issue in 1.7 and was fixed with PR #4093 but this not fixed in 2.2.x as the code shows

**Comment by traefiker** [comment]
Closed by #6827.
